src/plonk/prover.rs

```diff
@@ -152,15 +182,34 @@ impl<C: CurveAffine> Proof<C> {
 
             // Iterate over each wire of the permutation
             for (&(wire, _), permuted_wire_values) in wires.iter().zip(permuted_values.iter()) {
-                parallelize(&mut modified_advice, |modified_advice, start| {
-                    for ((modified_advice, advice_value), permuted_advice_value) in modified_advice
-                        .iter_mut()
-                        .zip(witness.advice[wire.0][start..].iter())
-                        .zip(permuted_wire_values[start..].iter())
-                    {
-                        *modified_advice *= &(x_0 * permuted_advice_value + &x_1 + advice_value);
+                match wire {
+                    Wire::Advice(wire) => {
+                        parallelize(&mut modified_advice, |modified_advice, start| {
+                            for ((modified_advice, advice_value), permuted_advice_value) in
+                                modified_advice
+                                    .iter_mut()
+                                    .zip(witness.advice[wire.0][start..].iter())
```

Comment:
You could put a `match` inside what you pass to `zip` here in order to consolidate all of these loops together.